Src/PerformanceCollector/Shared/QuickPulseTelemetryProcessor.cs

```diff
@@ -191,6 +264,112 @@ private static bool IsRequestSuccessful(RequestTelemetry request)
             return success.Value;
         }
 
+        private void ProcessTelemetry(ITelemetry telemetry)
+        {
+            // only process items that are going to the instrumentation key that our module is initialized with
+            if (this.config != null && !string.IsNullOrWhiteSpace(this.config.InstrumentationKey) && telemetry.Context != null
+                && string.Equals(telemetry.Context.InstrumentationKey, this.config.InstrumentationKey, StringComparison.OrdinalIgnoreCase))
+            {
+                var telemetryAsRequest = telemetry as RequestTelemetry;
+                var telemetryAsDependency = telemetry as DependencyTelemetry;
+                var telemetryAsException = telemetry as ExceptionTelemetry;
+
+                this.UpdateAggregates(telemetryAsRequest, telemetryAsDependency, telemetryAsException);
+
+                if (!this.disableFullTelemetryItems)
+                {
+                    this.CollectTelemetryDocuments(telemetryAsRequest, telemetryAsDependency, telemetryAsException);
+                }
+            }
+        }
+
+        private void CollectTelemetryDocuments(
+            RequestTelemetry telemetryAsRequest,
+            DependencyTelemetry telemetryAsDependency,
+            ExceptionTelemetry telemetryAsException)
```

Comment:
This is a performance optimization. This code is executed for every incoming request, so I want to avoid doing dynamic type conversions more than once.